src/tribler-core/tribler_core/modules/torrent_checker/torrent_checker.py

```diff
@@ -225,7 +246,7 @@ def update_torrents_checked(self, new_result):
         new_result_tuple = (new_result['infohash'], new_result['seeders'],
                             new_result['leechers'], new_result['last_check'])
         if new_result['seeders'] > 0:
-            self.torrents_checked.add(new_result_tuple)
+            self._torrents_checked.add(new_result_tuple)
```

Comment:
self._torrents_checked should probably be dict infohash -> (seeders, leechers, last_check) instead of a set, otherwise, multiple calls of update_torrents_checked can potentially create multiple records with the same infohash value and different seeders/leechers count

Reply:
Good point